go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -281,18 +292,12 @@ func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup, threads int
 			if err = c.iptablesManager.EnsureRule(utiliptables.TableMangle, utiliptables.ChainPrerouting, utiliptables.ProtocolIPv6, iptSaveMarkRule); err != nil {
 				return fmt.Errorf("failed to create rule in chain %s to save pkt marking: %v", utiliptables.ChainPrerouting, err)
 			}
-		}
-	}
-	if ovnconfig.Gateway.Mode == ovnconfig.GatewayModeLocal {
-		// If dst is a node IP, use main routing table and skip EIP routing tables
-		if err = c.ruleManager.Add(getNodeIPFwMarkIPRule()); err != nil {
-			return fmt.Errorf("failed to create IP rule for node IPs: %v", err)
-		}
-		// The fwmark of the packet is included in reverse path route lookup. This permits rp_filter to function when the fwmark is
-		// used for routing traffic in both directions.
-		stdout, _, err := util.RunSysctl("-w", "net.ipv4.conf.all.src_valid_mark=1")
-		if err != nil || stdout != "net.ipv4.conf.all.src_valid_mark = 1" {
-			return fmt.Errorf("failed to set sysctl net.ipv4.conf.all.src_valid_mark to 1")
+
+			// If dst is a node IP, use main routing table and skip EIP routing tables
+			// src_valid_mark is not applicable to ipv6
+			if err = c.ruleManager.Add(getNodeIPFwMarkIPRule(netlink.FAMILY_V6)); err != nil {
```

Comment:
is it possible to add a unit test for these pieces which is why they went unnoticed? cc @martinkennelly please correct me if I'm wrong cause I understand `Run` can be difficult to test.
so if you both tell me nah.. we are good I can merge this.

Reply:
Theres an e2e that covers this but it wasn't enabled and i didn't test it correctly (in LGW) :( 
I think thats sufficient.